models/Recording.ts

```diff
@@ -773,6 +776,23 @@ from (
   //------------------
   // INSTANCE METHODS
   //------------------
+  Recording.prototype.getNextState = function () {
+    const jobs = Recording.processingStates[this.type];
+    let nextState;
+    if (this.processingState == RecordingProcessingState.Reprocess) {
+      nextState = jobs[jobs.length - 1];
+    } else {
+      const job_index = jobs.indexOf(this.processingState);
+      if (job_index == -1) {
+        throw new Error(`Recording state unknown - ${this.processState}`);
```

Comment:
You need this error of job_index == -1 in your code otherwise it could do strange things.